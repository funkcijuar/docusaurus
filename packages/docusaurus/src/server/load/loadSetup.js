/**
 * Copyright (c) 2017-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import path from 'path';
import load from '../index';

// Helper methods to setup dummy/fake projects
const loadSetup = async name => {
  const fixtures = path.join(__dirname, '__tests__', '__fixtures__');
  const simpleSite = path.join(fixtures, 'simple-site');
  const customSite = path.join(fixtures, 'custom-site');

  switch (name) {
    case 'simple':
      return load(simpleSite);
    case 'custom':
      return load(customSite);
    default:
      return {};
  }
};

export default loadSetup;
