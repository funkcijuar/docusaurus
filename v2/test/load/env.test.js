/**
 * Copyright (c) 2017-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import path from 'path';
import loadEnv from '@lib/load/env';
import loadSetup from '../loadSetup';

describe('loadEnv', () => {
  test('website with both versioning & translation disabled', async () => {
    const {siteDir} = await loadSetup('simple');
    const siteConfig = {
      baseUrl: '/',
      organizationName: 'endiliey',
      projectName: 'hello',
      tagline: 'Hello World',
      title: 'Hello',
    };
    const env = loadEnv({siteDir, siteConfig});
    expect(env.versioning.enabled).toBe(false);
    expect(env.translation.enabled).toBe(false);
    expect(env).toMatchSnapshot();
  });

  test('website with versioning enabled', async () => {
    const {siteDir} = await loadSetup('versioned');
    const siteConfig = {
      baseUrl: '/',
      organizationName: 'endiliey',
      projectName: 'hello',
      tagline: 'Hello World',
      title: 'Hello',
    };
    const env = loadEnv({siteDir, siteConfig});
    expect(env.versioning.enabled).toBe(true);
    expect(env.translation.enabled).toBe(false);
    expect(env).toMatchSnapshot();
  });

  test('website with translation enabled', async () => {
    const {siteDir} = await loadSetup('translated');
    const siteConfig = {
      baseUrl: '/',
      organizationName: 'endiliey',
      projectName: 'hello',
      tagline: 'Hello World',
      title: 'Hello',
      defaultLanguage: 'en',
    };
    const env = loadEnv({siteDir, siteConfig});
    expect(env.versioning.enabled).toBe(false);
    expect(env.translation.enabled).toBe(true);
    expect(env).toMatchSnapshot();
  });

  test('website with versioning & translation enabled', async () => {
    const {siteDir} = await loadSetup('transversioned');
    const siteConfig = {
      baseUrl: '/',
      organizationName: 'endiliey',
      projectName: 'hello',
      tagline: 'Hello World',
      title: 'Hello',
      defaultLanguage: 'en',
    };
    const env = loadEnv({siteDir, siteConfig});
    expect(env.versioning.enabled).toBe(true);
    expect(env.translation.enabled).toBe(true);
    expect(env).toMatchSnapshot();
  });

  test('website with languages.js but no lang is enabled', () => {
    const siteDir = path.join(__dirname, '__fixtures__', 'bad-site');
    const siteConfig = {
      baseUrl: '/',
      organizationName: 'endiliey',
      projectName: 'hello',
      tagline: 'Hello World',
      title: 'Hello',
    };
    expect(() => {
      loadEnv({siteDir, siteConfig});
    }).toThrowErrorMatchingInlineSnapshot(
      `"Please at least enable one language in 'languages.js'"`,
    );
  });

  test('website with languages.js but no default language set', async () => {
    const {siteDir} = await loadSetup('translated');
    const siteConfig = {
      baseUrl: '/',
      organizationName: 'endiliey',
      projectName: 'hello',
      tagline: 'Hello World',
      title: 'Hello',
    };
    expect(() => {
      loadEnv({siteDir, siteConfig});
    }).toThrowErrorMatchingInlineSnapshot(
      `"Please set a default language in docusaurus.config.js which is enabled in languages.js"`,
    );
  });
});
