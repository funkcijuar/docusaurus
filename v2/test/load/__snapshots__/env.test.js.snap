// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadEnv website with both versioning & translation disabled 1`] = `
Object {
  "translation": Object {
    "defaultLanguage": Object {},
    "enabled": false,
    "enabledLanguages": Array [],
  },
  "versioning": Object {
    "defaultVersion": null,
    "enabled": false,
    "latestVersion": null,
    "versions": Array [],
  },
}
`;

exports[`loadEnv website with translation enabled 1`] = `
Object {
  "translation": Object {
    "defaultLanguage": Object {
      "enabled": true,
      "name": "English",
      "tag": "en",
    },
    "enabled": true,
    "enabledLanguages": Array [
      Object {
        "enabled": true,
        "name": "English",
        "tag": "en",
      },
      Object {
        "enabled": true,
        "name": "한국어",
        "tag": "ko",
      },
    ],
  },
  "versioning": Object {
    "defaultVersion": null,
    "enabled": false,
    "latestVersion": null,
    "versions": Array [],
  },
}
`;

exports[`loadEnv website with versioning & translation enabled 1`] = `
Object {
  "translation": Object {
    "defaultLanguage": Object {
      "enabled": true,
      "name": "English",
      "tag": "en",
    },
    "enabled": true,
    "enabledLanguages": Array [
      Object {
        "enabled": true,
        "name": "English",
        "tag": "en",
      },
      Object {
        "enabled": true,
        "name": "한국어",
        "tag": "ko",
      },
    ],
  },
  "versioning": Object {
    "defaultVersion": "1.0.1",
    "enabled": true,
    "latestVersion": "1.0.1",
    "versions": Array [
      "1.0.1",
      "1.0.0",
    ],
  },
}
`;

exports[`loadEnv website with versioning enabled 1`] = `
Object {
  "translation": Object {
    "defaultLanguage": Object {},
    "enabled": false,
    "enabledLanguages": Array [],
  },
  "versioning": Object {
    "defaultVersion": "1.0.1",
    "enabled": true,
    "latestVersion": "1.0.1",
    "versions": Array [
      "1.0.1",
      "1.0.0",
    ],
  },
}
`;
